### PR TITLE
[BPK-2427] Add direction in primary gradient

### PR DESCRIPTION
**Pull request by k0nserv**
All `BPKGradient` already contain a builtin in direction component in
their definiton. However when we render the primary gradient we
sometimes do so with an alternative direction. We want to be able to
theme primary gradients, while still supporting direction overrides for
specific cases.

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: Backpack/Gradient/Classes/BPKGradient.m [hunk 1]
```diff
@@ -184,6 +194,7 @@ + (CGPoint)endPointForDirection:(BPKGradientDirection)direction {
     case BPKGradientDirectionBottomRight:
         return CGPointMake(1.0, 1.0);
     default:
+        NSAssert(NO, @"Invalid `BPKGradientDirection` value %lu", (unsigned long)direction);
```
Comment (shaundon) [inline Backpack/Gradient/Classes/BPKGradient.m]: `%lu` is 'unsigned long' to save others a google 😇

File: Backpack/Gradient/Classes/BPKGradient.m [hunk 2]
```diff
@@ -61,6 +61,15 @@ - (instancetype)initWithColors:(NSArray<UIColor *> *)colors
     return self;
 }
 
+- (instancetype)cloneWithNewDirection:(BPKGradientDirection)newDirection {
```
Comment (georgegillams) [inline Backpack/Gradient/Classes/BPKGradient.m]: Why do we need `cloneWithNewDirection`. Would it make more sense for consumers to just clone a gradient and set the direction on the resulting view?

Comment (k0nserv) [reply Backpack/Gradient/Classes/BPKGradient.m]: I am confused, can you share some code examples?

Comment (georgegillams) [reply Backpack/Gradient/Classes/BPKGradient.m]: For example, why can't a user do something like this:
```
let gradient = Gradient();
let someOtherGradient = gradient.clone()
someOtherGradient.directionOverride = GradientDirectionTopLeft
```

Comment (georgegillams) [reply Backpack/Gradient/Classes/BPKGradient.m]: I would expect the API to allow me to just set the direction on the gradient instance

Comment (k0nserv) [reply Backpack/Gradient/Classes/BPKGradient.m]: This lets you do that, but it prefers a readonly API which is much nicer than a mutable API as you suggested

**Pull request merged**